inc/Engine/Optimization/RUCSS/Warmup/Scanner.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup;
+
+use WP_Rocket\Admin\Options;
+use WP_Rocket\Engine\Optimization\ContentTrait;
+
+class Scanner {
+	use ContentTrait;
+
+	/**
+	 * Background Process instance.
+	 *
+	 * @var ScannerProcess
+	 */
+	public $process;
+
+	/**
+	 * Options API instance.
+	 *
+	 * @var Options
+	 */
+	private $options_api;
+
+	/**
+	 * Items for which we get the resources to warmup.
+	 *
+	 * @var array
+	 */
+	public $items = [];
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param ScannerProcess $process Background process instance.
+	 * @param Options        $options_api Options API instance.
+	 */
+	public function __construct( ScannerProcess $process, Options $options_api ) {
+		$this->process     = $process;
+		$this->options_api = $options_api;
+	}
+
+	/**
+	 * Launches the scanner when activating the RUCSS option
+	 *
+	 * @since 3.9
+	 *
+	 * @param array $old_value Previous values for WP Rocket settings.
+	 * @param array $value     New values for WP Rocket settings.
+	 *
+	 * @return void
+	 */
+	public function start_scanner( $old_value, $value ) {
+		if (
+			! isset( $old_value['remove_unused_css'], $value['remove_unused_css'] )
```

Comment:
I think we removed the code of setting the RUCSS option value with the upgrade so the first time we don't have `$old_value['remove_unused_css']` so this will bail out here, right?